smarts/core/plan.py

```diff
@@ -64,7 +64,7 @@ def from_pose(cls, pose: Pose):
         )
 
 
-@dataclass(unsafe_hash=True)
+@dataclass(frozen=True, unsafe_hash=True)
```

Comment:
Why was this needed? I remember we removed `frozen=True` because it broke when running with Python 3.8

Reply:
It seems to work without issues somehow. I know the original issue was that you could not have a non-`frozen` dataclass inherit a `frozen` data class